tests/functional/test_warning.py

```diff
@@ -44,6 +45,10 @@ def test_flag_does_nothing_if_python_version_is_not_2(script):
     script.pip("list", "--no-python-version-warning")
 
 
+@pytest.mark.skipif(
+    sys.version_info >= (3, 10),
+    reason="distutils is deprecated in 3.10+"
+)
```

Comment:
This test was only added because the CPython test suite failed when packaging raised a deprecation warning (for legacy versions, IIRC). As ensurepip now forcibly silences deprecation warnings as part of the distutils removal, we could probably just drop this (after all, I don't think we should guarantee that pip runs warning-free as a general matter).

But it's no big deal either way.

Reply:
Gonna remove in a follow up. :)